src/theory/builtin/theory_builtin_rewriter.cpp

```diff
@@ -125,6 +135,30 @@ Node TheoryBuiltinRewriter::rewriteWitness(TNode node)
   return ElimShadowNodeConverter::eliminateShadow(node);
 }
 
+Node TheoryBuiltinRewriter::rewriteApplyIndexedSymbolic(TNode node)
+{
+  Assert(node.getNumChildren() > 1);
+  // if all arguments are constant, we return the non-symbolic version
+  // of the operator, e.g. (extract 2 1 #b0000) ---> ((_ extract 2 1) #b0000)
+  for (const Node& nc : node)
+  {
+    if (!nc.isConst())
+    {
+      return node;
+    }
+  }
+  Kind okind = node.getOperator().getConst<GenericOp>().getKind();
+  // determine how many arguments should be passed to the end function,
+  // for now, assume one
+  size_t nargs = 1;
```

Comment:
If we had the map of kind to number of indices as suggested above, this could be made more robust.

Reply:
This is different, this is the number of (term) arguments to the method.